btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go

```diff
@@ -0,0 +1,551 @@
+		metrics:                 metrics,
+		unbondingTracker:        NewTrackedDelegations(),
+		pendingTracker:          NewTrackedDelegations(),
+		unbondingDelegationChan: make(chan *newDelegation),
+		unbondingRemovalChan:    make(chan *delegationInactive),
+	}
+}
+
+func (sew *StakingEventWatcher) Start() error {
+	var startErr error
+	sew.startOnce.Do(func() {
+		sew.logger.Info("starting staking event watcher")
+
+		blockEventNotifier, err := sew.btcNotifier.RegisterBlockEpochNtfn(nil)
+		if err != nil {
+			startErr = err
+			return
+		}
+
+		// we registered for notifications with `nil` so we should receive best block immediately
+		select {
+		case block := <-blockEventNotifier.Epochs:
+			sew.currentBestBlockHeight.Store(uint32(block.Height))
+		case <-sew.quit:
+			startErr = errors.New("watcher quit before finishing start")
+			return
+		}
+
+		sew.logger.Infof("Initial btc best block height is: %d", sew.currentBestBlockHeight.Load())
+
+		sew.wg.Add(4)
+		go sew.handleNewBlocks(blockEventNotifier)
+		go sew.handleDelegations()
+		go sew.fetchDelegations()
+		go sew.handlerVerifiedDelegations()
+
+		sew.logger.Info("staking event watcher started")
+	})
+	return startErr
+}
+
+func (sew *StakingEventWatcher) Stop() error {
+	var stopErr error
+	sew.stopOnce.Do(func() {
+		sew.logger.Info("stopping staking event watcher")
+		close(sew.quit)
+		sew.wg.Wait()
+		sew.logger.Info("stopped staking event watcher")
+	})
+	return stopErr
+}
+
+func (sew *StakingEventWatcher) handleNewBlocks(blockNotifier *notifier.BlockEpochEvent) {
+	defer sew.wg.Done()
+	defer blockNotifier.Cancel()
+	for {
+		select {
+		case block, ok := <-blockNotifier.Epochs:
+			if !ok {
+				return
+			}
+			sew.currentBestBlockHeight.Store(uint32(block.Height))
+			sew.logger.Debugf("Received new best btc block: %d", block.Height)
+		case <-sew.quit:
+			return
+		}
+	}
+}
+
+// checkBabylonDelegations iterates over all active babylon delegations, and reports not already
+// tracked delegations to the unbondingDelegationChan
+func (sew *StakingEventWatcher) checkBabylonDelegations() error {
+	var i = uint64(0)
+	for {
+		delegations, err := sew.babylonNodeAdapter.BtcDelegations(i, sew.cfg.NewDelegationsBatchSize)
+
+		if err != nil {
+			return fmt.Errorf("error fetching active delegations from babylon: %v", err)
+		}
+
+		sew.logger.Debugf("fetched %d delegations from babylon", len(delegations))
+
+		for _, delegation := range delegations {
+			stakingTxHash := delegation.StakingTx.TxHash()
+
+			del := &newDelegation{
+				stakingTxHash:         stakingTxHash,
+				stakingTx:             delegation.StakingTx,
+				stakingOutputIdx:      delegation.StakingOutputIdx,
+				delegationStartHeight: delegation.DelegationStartHeight,
+				unbondingOutput:       delegation.UnbondingOutput,
+			}
+
+			// if we already have this delegation, skip it
+			if sew.unbondingTracker.GetDelegation(stakingTxHash) == nil && delegation.HasProof {
```

Comment:
hmm does this mean we won't be tracking unbodnign of delegations without inclusions proof ? 
If yes then we need to improve this, we need to be able to also report unbondings of staking transactions that do not have inclusion proof on babylon.